respy/pre_processing/model_checking.py

```diff
@@ -35,6 +31,46 @@ def validate_options(o):
         )
 
 
+def validate_params(params, optim_paras):
+    """Validate params."""
+    _validate_shocks(params, optim_paras)
+
+
+def _validate_shocks(params, optim_paras):
+    """Validate that the elements of the shock matrix are correctly sorted."""
+    choices = list(optim_paras["choices"])
+
+    if "shocks_sdcorr" in params.index:
+        sds_flat = [f"sd_{c}" for c in choices]
+
+        corrs_flat = []
+        for i, c_1 in enumerate(choices):
+            for c_2 in choices[: i + 1]:
```

Comment:
Isn't there an easier way to get all combinations permutations of two elements with itertools or a similar package?

Reply:
Due to the different labels for diagonal elements and lower triangular elements I would say no :). Using itertools would probably involve generating more than necessary and deleting some elements. I found this to be cleaner because labels are generated in the same order as required by the index.